examples/aws-sagemaker/endpoint.py

```diff
@@ -0,0 +1,219 @@
+
+        # Model
+        self.region = boto3.Session().region_name
+        self.acc = boto3.client("sts").get_caller_identity()["Account"]
+        self.repo_path = f"amazonaws.com/{self.repo_name}:latest"
+        self.image_uri = f"{self.acc}.dkr.ecr.{self.region}.{self.repo_path}"
+        self.image = [{"Image": self.image_uri}]
+        self.ROLE_ARN = role_arn
+
+        # Endpoint Configuration
+        self.variant_name = variant_name
+        self.model_name = model_name
+        self.initial_instance_count = instance_count
+        self.instance_type = instance_type
+        self.production_variants = [
+            {
+                "VariantName": self.variant_name,
+                "ModelName": self.model_name,
+                "InitialInstanceCount": self.initial_instance_count,
+                "InstanceType": self.instance_type,
+            }
+        ]
+        self.endpoint_config_name = config_name
+        self.endpoint_config = {
+            "EndpointConfigName": self.endpoint_config_name,
+            "ProductionVariants": self.production_variants,
+        }
+        self.endpoint_name = endpoint_name
+
+    def create_image(self):
+
+        subprocess.run(self.build_cmd, shell=True)
+
+    def create_ecr_repo(self):
+
+        try:
+            ecr = boto3.client("ecr", region_name=self.region_name)
+            ecr.create_repository(repositoryName=self.repo_name)
+
+        except ecr.exceptions.RepositoryAlreadyExistsException:
+            pass
+
+        repo_check = ecr.describe_repositories(repositoryNames=[self.repo_name])
+        pprint(repo_check["repositories"])
+
+    def push_image(self):
+
+        subprocess.call(["sh", self.push_script])
+
+    def create_model(self):
+
+        self.sm_boto3.create_model(
+            ModelName=self.model_name,
+            Containers=self.image,
+            ExecutionRoleArn=self.ROLE_ARN,
+            EnableNetworkIsolation=False,
+        )
+
+    def create_endpoint_config(self):
+
+        self.sm_boto3.create_endpoint_config(**self.endpoint_config)
+
+    def create_endpoint(self):
+
+        self.sm_boto3.create_endpoint(
+            EndpointName=self.endpoint_name,
+            EndpointConfigName=self.endpoint_config_name,
+        )
+
+        pprint(self.sm_boto3.describe_endpoint(EndpointName=self.endpoint_name))
+
+    def destroy_endpoint(self):
+
+        self.sm_boto3.delete_endpoint(EndpointName=self.endpoint_name)
+        self.sm_boto3.delete_endpoint_config(
+            EndpointConfigName=self.endpoint_config_name
+        )
+        self.sm_boto3.delete_model(ModelName=self.model_name)
+
+        print("endpoint and SageMaker model deleted")
+
+
+def parse_args():
+
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        dest="action",
+        type=str,
+        choices=["create", "destroy"],
+        help="Args to 'create' or 'destroy' a SageMaker endpoint.",
+    )
+
+    return parser.parse_args()
+
+
+def main():
```

Comment:
Can we remove argparse and change over to click, this will also alleviate my concerns below about the if else statement